File: 1d_eraser/1d.py
def main():
    t = int(input())
    for _ in range(t):
        n, k = map(int, input().split())
        s = input()
        i = 0
        num_moves = 0
        while (i < n):
            if s[i] == "B":
                num_moves += 1
                i += k - 1
            i += 1
        print(num_moves)
                
    
    
if __name__ == "__main__":
    main()

File: 1d_eraser/1d.cpp
#include <iostream>
#include <string>

using namespace std;

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n, k;
        cin >> n >> k;
        int num_moves = 0;
        string s;
        cin >> s;
        for (int j = 0; j < n; j++)
        {
            if (s[j] == 'B')
            {
                num_moves++;
                j += k - 1;
            }
        }

        cout << num_moves << "\n";

    }
}
